inhibit/inhibit.go

```diff
@@ -167,22 +168,30 @@ type InhibitRule struct {
 // NewInhibitRule returns a new InhibitRule based on a configuration definition.
 func NewInhibitRule(cr *config.InhibitRule) *InhibitRule {
 	var (
-		sourcem types.Matchers
-		targetm types.Matchers
+		sourcem labels.Matchers
+		targetm labels.Matchers
 	)
 
 	for ln, lv := range cr.SourceMatch {
-		sourcem = append(sourcem, types.NewMatcher(model.LabelName(ln), lv))
+		sourcem = append(sourcem, &labels.Matcher{
+			Type:  labels.MatchEqual,
+			Name:  ln,
+			Value: lv,
+		})
 	}
 	for ln, lv := range cr.SourceMatchRE {
-		sourcem = append(sourcem, types.NewRegexMatcher(model.LabelName(ln), lv.Regexp))
+		sourcem = append(sourcem, labels.NewRegexpMatcher(ln, lv.Regexp))
```

Comment:
I see what you are doing here. You are avoiding to recompile the regexp. I think this is not in a performance critical path, plus this will go away once the deprecated way of configuring inhibition rules is gone. So I suggest to not create a specialized `labels.NewRegexpMatcher` and do it as @aSquare14 did it before for the routing: https://github.com/prometheus/alertmanager/blob/master/dispatch/route.go#L105